Add the `sinon-chai` plugin to make test error messages more informative

At this moment, whenever we need to test that a spy or stub has been called once/twice/..., we do it the following way:

```js
it('should be called once', () => {
  const spy =  sinon.spy();
  ...
  expect(spy.calledOnce).to.be.true;
});
```

The downside is that, if such a test fails, the error message doesn't contain any information about how many times the spy has been actually called so that some extra debugging becomes required (especially when something fails in the CI):

```
AssertionError: expected false to be true
      + expected - actual

      -false
      +true
```

To make error messages more informative, there is [the `sinon-chai` plugin](https://www.chaijs.com/plugins/sinon-chai/) we might consider installing. 

See how the above error message would look like with the plugin installed:

```js
it('should be called once', () => {
  const spy =  sinon.spy();
  ...
  expect(spy).to.have.been.calledOnce; // Note, the notation is slightly different
});
```

... would show the following error message:

```
AssertionError: expected spy to have been called exactly once, but it was called twice
```

--- a/package.json
+++ b/package.json
@@ -6,26 +6,32 @@
   "license": "Apache-2.0",
   "main": "dist/index.js",
   "module": "dist/index.js",
+  "type": "module",
   "files": [
     "dist"
   ],
   "scripts": {
     "build": "tsc",
+    "build-plugins": "rollup -c rollup.config.js",
     "lint": "eslint src --ext .ts",
-    "prepublishOnly": "npm run build"
+    "prepublishOnly": "npm run build-plugins && npm run build"
   },
   "dependencies": {
     "@esm-bundle/chai": "^4.3.1",
     "@open-wc/semantic-dom-diff": "^0.19.7",
     "@polymer/polymer": "^3.5.1",
-    "lit": "^2.6.1"
+    "lit": "^2.6.1",
+    "sinon-chai": "3.7.0"
   },
   "devDependencies": {
+    "@rollup/plugin-commonjs": "^25.0.3",
+    "@rollup/plugin-node-resolve": "^15.1.0",
     "eslint": "^8.32.0",
     "eslint-config-prettier": "^8.6.0",
-    "eslint-plugin-prettier": "^4.2.1",
     "eslint-config-vaadin": "^1.0.0-alpha.10",
+    "eslint-plugin-prettier": "^4.2.1",
     "prettier": "^2.8.3",
+    "rollup": "^3.27.1",
     "typescript": "~4.9.4"
   }
 }

--- a/rollup.config.js
+++ b/rollup.config.js
@@ -1,0 +1,12 @@
+import commonjs from '@rollup/plugin-commonjs';
+import resolve from '@rollup/plugin-node-resolve';
+
+const plugins = ['sinon-chai'];
+
+export default plugins.map((plugin) => ({
+  input: plugin,
+  output: {
+    dir: 'dist/plugins'
+  },
+  plugins: [resolve(), commonjs()]
+}));

--- a/src/register-chai-plugins.ts
+++ b/src/register-chai-plugins.ts
@@ -1,4 +1,7 @@
 import { chaiDomDiff } from '@open-wc/semantic-dom-diff';
 import chai from '@esm-bundle/chai';
+// @ts-expect-error sinon-chai is bundled by Rollup
+import sinonChai from './plugins/sinon-chai.js';
 
 chai.use(chaiDomDiff);
+chai.use(sinonChai);

